server/src/main/java/org/apache/druid/server/coordinator/duty/KillUnusedSegments.java

```diff
@@ -166,4 +195,75 @@ private Interval findIntervalForKill(String dataSource)
     }
   }
 
+  private int getAvailableKillTaskSlots(@Nullable Double killTaskSlotRatio)
+  {
+    return Math.max(0, getKillTaskCapacity(killTaskSlotRatio) - getNumActiveKillTaskSlots());
+  }
+
+  private int getNumActiveKillTaskSlots()
+  {
+    final CloseableIterator<TaskStatusPlus> activeTasks =
+        FutureUtils.getUnchecked(overlordClient.taskStatuses(null, null, 0), true);
```

Comment:
I think this counts completed kill tasks against the limit, I used code similar to that introduced in https://github.com/apache/druid/pull/14581. So auto compaction I think has same bug

Reply:
This was fixed in https://github.com/apache/druid/pull/14667